src/components/inputs/rich-text-input/dropdown.js

```diff
@@ -0,0 +1,112 @@
+
+// eslint-disable-next-line
+const Label = props => <span>{props.children}</span>;
+
+const Dropdown = props => {
+  const DropdownItem = props.components.DropdownItem || StyledDropdownItem;
+  const DropdownLabel = props.components.DropdownLabel || Label;
+  return (
+    <Downshift
+      onChange={props.onChange}
+      selectedItem={props.value}
+      itemToString={headings => (headings ? headings.label : '')}
+    >
+      {({
+        isOpen,
+        toggleMenu,
+        getToggleButtonProps,
+        getItemProps,
+        highlightedIndex,
+        selectedItem,
+      }) => {
+        const { onClick, ...restOfToggleButtonProps } = getToggleButtonProps();
+        return (
+          <div>
+            <Tooltip
+              title="Text styles"
+              placement="bottom"
+              off={isOpen}
+              style={{ body: { zIndex: 9999 } }}
+            >
+              <AccessibleButton
+                label={props.label}
+                css={getButtonStyles()}
+                {...restOfToggleButtonProps}
+                onMouseDown={toggleMenu}
+              >
+                <DropdownLabel>{props.label}</DropdownLabel>
+              </AccessibleButton>
+            </Tooltip>
+            <div style={{ position: 'relative' }}>
+              {isOpen ? (
+                <DropdownContainer>
+                  {props.options.map((item, index) => {
+                    const {
+                      onClick: dropdownOnClick,
+                      ...restOfItemProps
+                    } = getItemProps({
+                      key: index,
+                      index,
+                      item,
+                    });
+                    return (
+                      // eslint-disable-next-line react/jsx-key
+                      <DropdownItem
+                        {...restOfItemProps}
+                        onMouseDown={event => {
+                          restOfItemProps.onMouseDown(event);
+                          dropdownOnClick(event);
+                        }}
+                        value={item.value}
+                        isSelected={item.value === selectedItem}
+                        isHovered={highlightedIndex === index}
+                      >
+                        {item.label}
+                      </DropdownItem>
+                    );
+                  })}
+                </DropdownContainer>
+              ) : null}
+            </div>
+          </div>
+        );
+      }}
+    </Downshift>
+  );
+};
+
+Dropdown.displayName = 'Dropdown';
+
+Dropdown.propTypes = {
+  components: PropTypes.shape({
+    DropdownItem: PropTypes.func,
+    DropdownLabel: PropTypes.func,
+  }),
+  label: PropTypes.string,
+  value: PropTypes.string,
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      label: PropTypes.string,
+      value: PropTypes.string,
+    })
+  ).isRequired,
+  onChange: PropTypes.func.isRequired,
+};
+
+Dropdown.defaultProps = {
```

Comment:
Ah sorry, Forget comment above. Side note: checking for it instead of defaulting to `{}` might still be easier-ish.